File: frontend/src/types/system.ts
import type { FacebookConnectionStatus } from "./facebook";

export type PublishMode = "dry_run" | "facebook_schedule";

export interface HealthStatus {
  status: "ok";
  service: string;
}

export interface FacebookConfigurationStatus {
  page_id_configured: boolean;
  access_token_configured: boolean;
  fully_configured: boolean;
}

export interface SystemStatus {
  application_mode: string;
  authentication_required: boolean;
  supabase_configured: boolean;
  publish_mode: PublishMode;
  automation_enabled: boolean;
  publishing_enabled: boolean;
  timezone: string;
  graph_api_version: string;
  facebook: FacebookConfigurationStatus;
}

export interface SystemSnapshot {
  health: HealthStatus;
  status: SystemStatus;
  facebook: FacebookConnectionStatus;
}

File: backend/app/api/system.py
"""Safe, non-secret system configuration status."""

from fastapi import APIRouter
from pydantic import BaseModel

from app.config import PublishMode, get_settings


router = APIRouter(prefix="/api/system", tags=["system"])


class FacebookConfigurationStatus(BaseModel):
    page_id_configured: bool
    access_token_configured: bool
    fully_configured: bool


class SystemStatusResponse(BaseModel):
    application_mode: str
    publish_mode: PublishMode
    automation_enabled: bool
    publishing_enabled: bool
    timezone: str
    graph_api_version: str
    facebook: FacebookConfigurationStatus


@router.get("/status", response_model=SystemStatusResponse)
async def system_status() -> SystemStatusResponse:
    settings = get_settings()
    page_id_configured = settings.facebook_page_id_configured
    access_token_configured = settings.facebook_token_configured

    return SystemStatusResponse(
        application_mode=settings.application_mode,
        publish_mode=settings.publish_mode,
        automation_enabled=settings.automation_enabled,
        publishing_enabled=settings.publishing_enabled,
        timezone=settings.app_timezone,
        graph_api_version=settings.facebook_graph_api_version,
        facebook=FacebookConfigurationStatus(
            page_id_configured=page_id_configured,
            access_token_configured=access_token_configured,
            fully_configured=page_id_configured and access_token_configured,
        ),
    )
